blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -65,20 +70,20 @@ All joins provided by Kafka Streams were designed with aggregation in mind, like
 - (A+C, null) <== we will never have (A+B+C)
 ```
 
-In our case, we are building a state machine and want to keep the last state of execution, meaning we do not want to see the intermediate states. In this case, we have no choice but to build a custom joiner since Kafka Streams doesn't have one built-in.
+However, we are building a state machine and want to keep the last state of execution, meaning we do not want to see the intermediate states. In this case, we have no choice but to build a custom joiner since Kafka Streams doesn't have one built-in.
 
 Our custom joiner needs to:
 - [Manually create a store](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorFromExecutionTransformer.java) that will save the last state of an execution.
 - Create a custom [merge function](https://github.com/kestra-io/kestra/blob/8a17519b5b41e8ba0ab8e0044c9c6b1e887ccd94/runner-kafka/src/main/java/io/kestra/runner/kafka/executors/ExecutorMain.java#L216-L246) that will merge the execution stream with the task results stream.
 - [Get the last value](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorJoinerTransformer.java) from the state, add the task result and emit the new state that will finally be saved on the store and the final topics.
```

Comment:
Is it okay to rephase to "inally be saved on the store and the final topic."? Or are there multiple final topics?

Reply:
There is only one topic on that particular use case